dax/api.go

```diff
@@ -963,3 +402,16 @@ func (d *Dax) Close() error {
 	}
 	return nil
 }
+
+func (d *Dax) ContextWithTimeout(ctx context.Context) (context.Context, context.CancelFunc) {
+	// avoid nil context here
+	if ctx == nil {
+		ctx = context.Background()
+	}
+	if d.config.RequestTimeout > 0 {
+		return context.WithTimeout(ctx, d.config.RequestTimeout)
+	}
```

Comment:
This would mean that we would never honour timeout set in `ctx` right? We would only honour the timeout that's set in `d.config.RequestTimeout`.

For v2, we are thinking to eliminate `d.config.RequestTimeout` and just rely on the timeout set in `ctx`. What are your thoughts on this?

Please refer to https://github.com/aws/aws-dax-go/issues/46#issuecomment-1584937167 for discussion around this topic.

Reply:
No, this code will always set d.config.RequestTimeout as timeout in context. A timeout in the parent context is also valid.
https://go.dev/play/p/XfE67i2BVCD

If you want to remove d.config.RequestTimeout, you can do so but setting timeout is completely up to the user.
I do not recommend it. We should always have timeout set to the default value recommended by aws unless set by user.

The problem with https://github.com/aws/aws-dax-go/issues/46#issuecomment-1584937167 is that the RequestTimeout is not set unless the context is nil right?

Reply:
So what we are saying here is that both values will be read but if the user passes a `ctx` with a `timeout`, this will override `config.RequesTimeout`?

Reply:
@anudeeb 
I thought about this again.

It may be better to let user set context.timeout. This is because the amount of time it takes depends on kinds of the data processing. For example, fetching or counting lots of data may be different than fetching single record.

so I think it is good thoughts to eliminate `config.RequestTimeout`

Reply:
changed in https://github.com/aws/aws-dax-go/pull/43/commits/eaf620e2642f601d015a0b1b89fcff3e7a6372bc